src/OpenFeature/Api.cs

```diff
@@ -265,5 +265,37 @@ internal void AddClientHandler(string client, ProviderEventTypes eventType, Even
 
         internal void RemoveClientHandler(string client, ProviderEventTypes eventType, EventHandlerDelegate handler)
             => this._eventExecutor.RemoveClientHandler(client, eventType, handler);
+
+        /// <summary>
+        /// Update the provider state to READY and emit an READY after successful init.
+        /// </summary>
+        private void afterInitialization(FeatureProvider provider)
+        {
+            provider.Status = ProviderStatus.Ready;
+            var eventPayload = new ProviderEventPayload
+            {
+                Type = ProviderEventTypes.ProviderReady,
+                Message = "Provider initialization complete",
+                ProviderName = provider.GetMetadata().Name,
+            };
+
+            this._eventExecutor.EventChannel.Writer.WriteAsync(new Event { Provider = provider, EventPayload = eventPayload });

```

Comment:
I'm not a fan of this line, to be honest. We have an async method here that has yet to be awaited. The `WriteAsync` returns a `ValueTask`, which can lead to performance degradation in this case. Check: https://learn.microsoft.com/en-gb/dotnet/fundamentals/code-analysis/quality-rules/ca2012

Reply:
I'm not sure... 

From your article:

> Attempts to consume a ValueTask multiple times or to directly access one's result before it's known to be completed may result in an exception or corruption.

This makes sense, and I'm aware of the pitfalls here.

> Ignoring such a ValueTask is likely an indication of a functional bug and may degrade performance.

I'm not sure if this is saying that not awaiting a task at all (using it in a "fire and forget" way) is a problem. I can't see how it would be, but I could be missing something... I can use a combination of `WaitToWriteAsync` and `TryWrite`, but I don't think the end result is much different.

Reply:
In this case, the `ValueTask` is not used again, so the code will not directly cause any issues. However, I am more worried if we need to reuse the value in the future. Thanks for the change 😄